src/Makefile

```diff
@@ -3,8 +3,30 @@
 PREFIX ?= /usr/local
 BINDIR ?= $(PREFIX)/bin
 LIBDIR ?= $(PREFIX)/lib
+ARCH ?= x64
 CC ?= gcc
 
+TARGET_MACHINE := $(shell $(CC) -dumpmachine)
+
+ifneq (,$(findstring aarch64-,$(TARGET_MACHINE)))
+	ARCH := arm64
+else ifneq (,$(findstring arm-,$(TARGET_MACHINE)))
+	ARCH := arm
+else ifneq (,$(findstring mips-,$(TARGET_MACHINE)))
+	ARCH := mips
+else ifneq (,$(findstring x86_64-,$(TARGET_MACHINE)))
+	ARCH := x64
+else ifneq (,$(findstring i686-,$(TARGET_MACHINE)))
+	ARCH := x86
+else ifneq (,$(findstring i586-,$(TARGET_MACHINE)))
+	ARCH := x86
+else ifneq (,$(findstring i486-,$(TARGET_MACHINE)))
+	ARCH := x86
+else ifneq (,$(findstring i386-,$(TARGET_MACHINE)))
+	ARCH := x86
+endif
+
```

Comment:
Also, if the default value for `$(ARCH)` is empty, we could have a check here:

```make
ifeq ($(strip $(ARCH)),)
$(error Unsupported target CPU: $(TARGET_MACHINE))
endif
```

This way we would be providing a helpful error message, instead of trying to make a build which would mysteriously fail to link. That could happen when the value of `$(TARGET_MACHINE)` is not matched against any known value, for example with PowerPC as target, and then we would have some `.o` ELF objects for the (assumed) `x86_64`, and linking step would fail because the rest of the ELF objects were built for another architecture (PowerPC in the example).